src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,134 @@
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string;
+
     @Override
     public String getString() {
-        return null;
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (string == null)
+            return 0;
+        if (string.length() == 0)
+            return 0;
+
+        int count = 0;
+        for (int i = 0; i < string.length() - 2; i++) {
+            if (i == 0 && !isSpaceChar(string.charAt(i)))
+                count++;
+            else if (isSpaceChar(string.charAt(i)) && !isSpaceChar(string.charAt(i + 1)))
+                count++;
+        }
+
+        return count;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0)
+            throw new IllegalArgumentException();
+        if (n > string.length())
+            throw new IndexOutOfBoundsException();
+
+        StringBuilder newStr = new StringBuilder();
```

Comment:
StringBuilder in retrospect feels a lot simpler than the solution I came up with, atleast in this case haha